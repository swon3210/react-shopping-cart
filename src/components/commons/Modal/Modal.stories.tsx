import BackDrop from '../BackDrop/BackDrop';
import Modal, { Props } from './Modal';

export default {
  title: 'Components/Commons/Modal',
  component: Modal,
};

const Template = (args: Props) => <Modal {...args} />;

export const Default = Template.bind({});
(Default as any).args = {
  zIndex: 100,
};
