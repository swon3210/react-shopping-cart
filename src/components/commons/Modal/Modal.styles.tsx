import styled, { keyframes } from 'styled-components';

const fadeIn = keyframes`
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
`;

export const Modal = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  top: 0;
  width: 100vw;
  height: 100vh;
  z-index: 100;
  animation: ${fadeIn} 0.5s forwards;
`;

export const ModalContent = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 4px;
  box-shadow: 5px 5px 25px rgba(0, 0, 0, 0.1);
  background-color: ${({ theme }) => theme.WHITE};
`;
