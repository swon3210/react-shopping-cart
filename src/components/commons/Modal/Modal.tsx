import * as Styled from './Modal.styles';
import BackDrop from '../BackDrop/BackDrop';

export interface Props {
  animation?: 'fadeIn';
  children: React.ReactNode;
  onBackDropClick?: () => void;
}

const Modal = ({ children, animation = 'fadeIn', onBackDropClick }: Props) => {
  return (
    <Styled.Modal animation={animation}>
      <BackDrop onBackDropClick={onBackDropClick} />
      <Styled.ModalContent>{children}</Styled.ModalContent>
    </Styled.Modal>
  );
};

export default Modal;
