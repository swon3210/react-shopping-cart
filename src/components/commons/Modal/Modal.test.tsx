import Modal from './Modal';
import { render } from '@testing-library/react';

describe('Modal Component', () => {
  it('Modal Snapshot', () => {
    const modalUtil = render(<Modal>TEST</Modal>);
    expect(modalUtil.asFragment()).toMatchSnapshot();
  });
});
