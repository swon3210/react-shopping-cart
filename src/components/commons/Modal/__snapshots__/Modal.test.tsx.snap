// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component Modal Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvaa bRMtVR"
  >
    <div
      class="sc-dkPtyc iVayro"
    />
    <div
      class="sc-gsDJrp dQYwbU"
    >
      TEST
    </div>
  </div>
</DocumentFragment>
`;
